DiagnosticsExtension/Models/ConnectionStringValidator/ServiceBusValidator.cs

```diff
@@ -118,6 +120,157 @@ protected async Task<TestConnectionData> TestConnectionStringAsync(string connec
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.Forbidden;
+                }
+                else
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.UnknownError;
+                }
+                response.Exception = e;
+            }
+
+            return response;
+        }
+        protected async Task<TestConnectionData> TestConnectionStringViaAppSettingAsync(string appSettingName, string entityName)
+        {
+            string value, appSettingClientIdValue, appSettingClientCredValue = "";
+            ServiceBusClient client = null;
+            var envDict = Environment.GetEnvironmentVariables();
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
+                {
+                    value = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new ServiceBusClient(value);
+                }
+                catch (Exception e)
+                {
+                    throw new MalformedConnectionStringException(e.Message, e);
+                }
+
+            }
+            else
+            {
+                try
+                {
+                    value = Environment.GetEnvironmentVariable(appSettingName + "__fullyQualifiedNamespace");
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + "__clientId");
+                        appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + "__credential");
+                        if (!string.IsNullOrEmpty(appSettingClientIdValue))
+                        {
+                            if (appSettingClientCredValue != "managedidentity")
+                            {
+
+                                throw new ManagedIdentityException("managedidentitymissed");
+                            }
+                            else
+                            {
+                                identityType = ConnectionStringValidationResult.ManagedIdentityType.User;
+                                client = new ServiceBusClient(value, new Azure.Identity.ManagedIdentityCredential(appSettingClientIdValue));
+                            }
+                        }
+                        else
+                        {
+                            identityType = ConnectionStringValidationResult.ManagedIdentityType.System;
+                            client = new ServiceBusClient(value, new Azure.Identity.ManagedIdentityCredential());
+                        }
+                    }
+                    else
+                    {
+                        throw new ManagedIdentityException("fullyQualifiedNamespacemissed");
+                    }
+                }
+                catch (Exception e)
```

Comment:
why wrap the exception again here?  just let it go up the stack like the code below.  e.g. exception from line 255 needlessly will get wrapped in the exception thrown here.

Reply:
in the exception in line 255, we are clear it is because of fullyQualifiedNamespace missing. 
in the exception below in line 260, the exception would be due to ManagedIdentity but it is not straight away related to fullyQualifiedNamespace. It may be due to unable to generate token kind of things

Reply:
my point is, just let the exceptions flow through to ValidateViaAppsettingAsync where we are systematically handling all exceptions regardless of where in the stack they are thrown (our code or SDK code)

Reply:
I don't see the feedback here addressed yet.

Reply:
fixed

Reply:
I still don't see the feedback addressed - can we discuss this point over a call as it seems for this point, I am unable to communicate the feedback clearly via PR feedback.